src/ray/common/memory_monitor.cc

```diff
@@ -185,6 +185,46 @@ std::tuple<int64_t, int64_t> MemoryMonitor::GetLinuxMemoryBytes() {
   return {used_bytes, mem_total_bytes};
 }
 
+int64_t MemoryMonitor::GetProcessMemoryBytes(int64_t process_id) {
+  std::stringstream ss;
+  ss << "/proc/" << std::to_string(process_id) << "/smaps_rollup";
+  return GetLinuxProcessMemoryBytesFromSmap(ss.str());
+}
+
+/// TODO:(clarng) align logic with psutil / Python-side memory calculations
+int64_t MemoryMonitor::GetLinuxProcessMemoryBytesFromSmap(const std::string smap_path) {
```

Comment:
Should we try catch the entier code just in case?

Reply:
And return kNull if there's any exception (since Ray has no exception code base, we should avoid thworing an exception to the upper layer)